File: 1st Month/Home Tasks/8. DataOperation_no_1.R
data = read.csv(file = file.choose()) # Will open Dialog box to select
data = read.csv("G:\\Courses\\NAVVTAC\\1st Month\\Data Files\\dat.csv")


library("dplyr")

# ==============|1. Count Numbers of records in each group |==============



data_ctry <- data %>% 
  group_by(Country) %>% 
  count
data_ctry


# --------| Crop |--------


data_crop <- data %>% 
  group_by(Crop) %>% 
  summarise(Count=n())



# --------| Information |--------


data_info <- data %>% 
  group_by(Information) %>% 
  summarise(Count=n())



# --------| Year |--------


data_year <- data %>% 
  filter(
    Year >= 2000
  ) %>% 
  group_by(Year) %>% 
  summarise(Count=n())



# --------| Source |--------


data_src <- data %>% 
  group_by(Source) %>% 
  summarise(Count=n())


# ==============|2. Summarise by Mean yeild and year Range  |==============


data_mean <- data %>% 
  filter(
    Year >= 2000 & Year <= 2005
  ) %>% 
  group_by(Crop) %>% 
  summarise(Yeild_mean = mean(Value))


# ==============|3. Summarise by Crop  |==============


# --------| Value |--------


data_min <- data %>% 
  group_by(Crop) %>% 
  summarise(min_val = min(Value), max_val = max(Value))


# --------| Year |--------


data_min <- data %>% 
  group_by(Crop) %>% 
  summarise(min_year = min(Year), max_year = max(Year))



# ==============|3. Mutate Column  |==============


data_mut <- data %>% 
  mutate(
    ctry = ifelse(Country == "United States of America","USA",Country)
  )



# ==============|4. Sorting  |==============



data_sort <- data[order(data$Crop,data$Value),]

File: 1st Month/Home Tasks/8. DataOperation_no_1.py
import numpy as np
import pandas as pd

data = pd.read_csv("G:\\Courses\\NAVVTAC\\1st Month\\Data Files\\dat.csv")

# # ==============|1. Count Numbers of records in each group |==============


# --------| Crop |--------

data_ctry = data.groupby('Country').size().reset_index(name='count')

# --------| Crop |--------

data_crop = data.groupby('Crop').size().reset_index(name='Count')

# --------| Information |--------

data_info = data.groupby('Information').size().reset_index(name='Count')

# --------| Year |--------

data_year = data[data['Year'] >= 2000].groupby('Year').size().reset_index(name='Count')

# --------| Source |--------

data_src = data.groupby('Source').size().reset_index(name='Count')

# ==============|2. Summarise by Mean yield and year Range |==============

data_mean = data[(data['Year'] >= 2000) & (data['Year'] <= 2005)].groupby('Crop')['Value'].mean().reset_index(name='Yeild_mean')



# ==============|3. Summarise by Crop  |==============


# --------| Value |--------

data_crop__val = data.groupby('Crop').agg(
  min_val = ("Value",min),
  max_val = ("Value",min),
)


# --------| Year |--------

data_crop_year = data.groupby('Crop').agg(
  min_val = ("Year",min),
  max_val = ("Year",min),
)



# ==============|3. Mutate Column  |==============


data["ctry"] = np.where(
  data["Country"] == "United States of America",
  "USA",
  data["Country"]
)



# ==============|4. Sorting  |==============

data.sort_values(by=["Crop","Value"])


